fuzz/dex/v0.1/fuzzDex_test.go

```diff
@@ -0,0 +1,274 @@
+		require.Nil(t, err)
+
+	case re.WithProbability(pfe.addLiquidityProb):
+
+		swapPair := getRandomSwapPair(r, pfe)
+		seed := r.Intn(pfe.addLiquidityMaxValue) + 1
+		amountA := seed
+		amountB := seed
+		amountAmin := seed / 100
+		amountBmin := seed / 100
+
+		err := pfe.addLiquidity(user, swapPair, amountA, amountB, amountAmin, amountBmin, statistics)
+		require.Nil(t, err)
+
+	case re.WithProbability(pfe.swapProb):
+
+		swapPair := getRandomSwapPair(r, pfe)
+		fixedInput := false
+		amountA := 0
+		amountB := 0
+		fixedInput = r.Intn(2) != 0
+		seed := r.Intn(pfe.swapMaxValue) + 1
+		amountA = seed
+		amountB = seed / 100
+
+		if fixedInput {
+			err := pfe.swapFixedInput(user, swapPair, amountA, amountB, statistics)
+			require.Nil(t, err)
+		} else {
+			err := pfe.swapFixedOutput(user, swapPair, amountA, amountB, statistics)
+			require.Nil(t, err)
+		}
+
+	case re.WithProbability(pfe.queryPairsProb):
+
+		swapPair := getRandomSwapPair(r, pfe)
+		err := pfe.checkPairViews(user, swapPair, statistics)
+		require.Nil(t, err)
+
+	case re.WithProbability(pfe.enterFarmProb):
+
+		amount := r.Intn(pfe.enterFarmMaxValue) + 1
+		farm := getRandomFarm(r, pfe)
+		err := pfe.enterFarm(user, farm, amount, statistics)
+		require.Nil(t, err)
+
+	case re.WithProbability(pfe.exitFarmProb):
+
+		amount := r.Intn(pfe.exitFarmMaxValue) + 1
+		err := pfe.exitFarm(amount, statistics, r)
+		require.Nil(t, err)
+
+	case re.WithProbability(pfe.claimRewardsProb):
+
+		amount := r.Intn(pfe.claimRewardsMaxValue) + 1
+		err := pfe.claimRewards(amount, statistics, r)
+		require.Nil(t, err)
+
+	case re.WithProbability(pfe.increaseBlockNonceProb):
+
+		err := pfe.increaseBlockNonce(pfe.blockNonceIncrease)
+		require.Nil(t, err)
+	default:
+	}
+}
+
+func getRandomSwapPair(r *rand.Rand, pfe *fuzzDexExecutor) SwapPair {
+	seed := r.Intn(2)
+	randomPair := SwapPair{}
+
+	if seed == 0 {
+		randomPair.address = pfe.wemeSwapAddress
+		randomPair.lpToken = pfe.wemeLpTokenId
+		randomPair.firstToken = pfe.wegldTokenId
+		randomPair.secondToken = pfe.mexTokenId
+	} else {
+		randomPair.address = pfe.webuSwapAddress
+		randomPair.lpToken = pfe.webuLpTokenId
+		randomPair.firstToken = pfe.wegldTokenId
+		randomPair.secondToken = pfe.busdTokenId
+	}
+
+	return randomPair
+}
+
+func getRandomFarm(r *rand.Rand, pfe *fuzzDexExecutor) Farm {
+	seed := r.Intn(3)
+	randomFarm := Farm{}
+
+	if seed == 0 {
+		randomFarm.address = pfe.wemeFarmAddress
+		randomFarm.farmToken = pfe.wemeFarmTokenId
+		randomFarm.farmingToken = pfe.wemeLpTokenId
+		randomFarm.rewardToken = pfe.mexTokenId
+	} else if seed == 1 {
+		randomFarm.address = pfe.webuFarmAddress
+		randomFarm.farmToken = pfe.webuFarmTokenId
+		randomFarm.farmingToken = pfe.webuLpTokenId
+		randomFarm.rewardToken = pfe.mexTokenId
+	} else {
+		randomFarm.address = pfe.mexFarmAddress
+		randomFarm.farmToken = pfe.mexFarmTokenId
+		randomFarm.farmingToken = pfe.mexTokenId
+		randomFarm.rewardToken = pfe.mexTokenId
+	}
```

Comment:
Would it be possible to define a field `pfe.Farms` of type `[]*Farm`, and each index in this array is a predefined constant? This should simplify `getRandomSwapPair()` and `getRandomFarm()`. In their current form, it is not obvious that the `rewardToken` is always `mexTokenId`, nor that `firstToken` is always `wegldTokenId`.

Reply:
done